demonstrations/tutorial_adversarial_attacks_QML.py

```diff
@@ -0,0 +1,470 @@
+# amongst others, for malware detection or the prevention of attacks on networks. Given their critical
+# roles, any successful attack of these ML models can lead to severe consequences, ranging from
+# financial losses and privacy violations to threats to human life. As we have seen above, adversarial
+# attacks are imperceptible to humans, and hence difficult to detect. For this reason, it is essential
+# that ML models in security-critical applications are robust against these types of attacks.
+#
+# :doc:`Quantum machine learning (QML) </whatisqml>` has been shown to have theoretical advantages over classical ML methods [#Liu]_
+# and is becoming increasingly popular. However, first works in this direction suggest that QML
+# suffers from the same vulnerabilities as classical ML [#Lu]_. How the vulnerability of QML models
+# relates to classical models and how robust the models are in comparison is evaluated in [#Wendlinger]_. But
+# enough talk, let’s see how we can actually attack a QML model!
+#
+
+######################################################################
+# Let’s see this in action!
+# -------------------------
+#
+# Setting up the environment
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# For this tutorial, we will use the PennyLane :class:`~pennylane.qnn.TorchLayer` class to perform circuit operations
+# and optimizations with the PyTorch backend. Thus, we need to import the torch library alongside
+# PennyLane:
+#
+
+import pennylane as qml
+from pennylane import numpy as np
+import torch
+from matplotlib import pyplot as plt
+
+######################################################################
+# Visualization of the dataset
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# As in the paper [#Wendlinger]_, we make use of the `PlusMinus <pennylane.ai/datasets/other/plus-minus>`_ dataset (available via `PennyLane Datasets <https://pennylane.ai/datasets/>`_), which serves as a good baseline for
+# evaluating a QML image classification model’s ability to find useful features in the input. It also
+# allows us to define the usefulness of attacks on the QML model while being low-dimensional enough to
+# perform scalable training (more info on the dataset can be found in [#Wendlinger]_). It consists of four
+# classes of :math:`16\times16` pixel grayscale images which show one of the four symbols
+# :math:`\{+,-,\vdash,\dashv\}`. Below we visualize one sample of each class to get an understanding
+# of the dataset.
+#
+# The data can be loaded directly from `PennyLane Datasets <https://pennylane.ai/datasets/>`_ for easy integration into PennyLane circuits and optimization code.
+#
+
+# we can use the dataset hosted on PennyLane
+# pm = qml.data.load('other, name='PlusMinus')
+
+pm = qml.data.Dataset()
+pm.read("./adversarial_attacks/data/PlusMinus.h5")
```

Comment:
Updating to use the hosted dataset:

```suggestion
[pm] = qml.data.load('other', name='plus-minus')
```